src/cpp/security/accesscontrol/Permissions.cpp

```diff
@@ -346,6 +346,46 @@ static bool rfc2253_string_compare(
 }
 
 // Auxiliary functions
+static BIO* load_and_verify_document(
+        X509_STORE* store,
+        BIO* in,
+        SecurityException& exception)
+{
+    BIO* out = nullptr;
+
+    assert(nullptr != store);
+    assert(nullptr != in);
+
+    BIO* indata = nullptr;
+    PKCS7* p7 = SMIME_read_PKCS7(in, &indata);
+    if (nullptr == p7)
+    {
+        exception = _SecurityException_("Cannot read as PKCS7 the file ");
+        return nullptr;
```

Comment:
Don't we need to free `indata` before returning here??

Reply:
I don't think so, since it is an output argument. See [the docs](https://www.openssl.org/docs/man1.0.2/man3/SMIME_read_PKCS7.html) for more info.